app/screens/apps_form/apps_form_component.tsx

```diff
@@ -153,24 +144,58 @@ function AppsFormComponent({
             undefined,
             intl.formatMessage({id: 'interactive_dialog.submit', defaultMessage: 'Submit'}),
         );
-        base.enabled = submitting;
+        base.enabled = !submitting;
         base.showAsAction = 'always';
         base.color = theme.sidebarHeaderTextColor;
         return base;
-    }, [theme.sidebarHeaderTextColor, intl, Boolean(submitButtons)]);
+    }, [theme.sidebarHeaderTextColor, Boolean(submitButtons), submitting, intl]);
 
     useEffect(() => {
         setButtons(componentId, {
             rightButtons: rightButton ? [rightButton] : [],
         });
-    }, [rightButton, componentId]);
+    }, [componentId, rightButton]);
 
     useEffect(() => {
         const icon = CompassIcon.getImageSourceSync('close', 24, theme.sidebarHeaderTextColor);
         setButtons(componentId, {
             leftButtons: [makeCloseButton(icon)],
         });
-    }, [theme]);
+    }, [componentId, theme]);
+
+    const updateErrors = useCallback((elements: DialogElement[], fieldErrors?: {[x: string]: string}, formError?: string): boolean => {
+        let hasErrors = false;
+        let hasHeaderError = false;
+        if (formError) {
+            hasErrors = true;
+            hasHeaderError = true;
+            setError(formError);
+        }
+
+        if (fieldErrors && Object.keys(fieldErrors).length > 0) {
```

Comment:
This would be nice in a validation function, but I see this code was this way before, so I wouldn't do it now.